pkg/lib/config/identity.go

```diff
@@ -320,6 +320,10 @@ func (c OAuthSSOProviderConfig) Alias() string {
 	return ""
 }
 
+func (c OAuthSSOProviderConfig) Type() string {
+	return c["type"].(string)
+}
+
```

Comment:
Do not need to add this method. You write this instead

```
c.AsProviderConfig().Type()
```

Reply:
so many types representing the same thing..
When do we use `oauthrelyingparty.ProviderConfig` and when do we use `OAuthSSOProviderConfig`? How do we draw the line?

Reply:
> When do we use oauthrelyingparty.ProviderConfig

When you need to use the fields defined in oauthrelyingparty.ProviderConfig

> when do we use OAuthSSOProviderConfig

When you need to use the fields NOT defined in oauthrelyingparty.ProviderConfig

> so many types representing the same thing..

Underlying it is just a dictionary. We have a reason to do that, but that requires a explanation that you may not be interested in knowing.

Reply:
🤡